File: webmvp/src/lib/playlistInviteToken.ts
const TOKEN_ALPHABET =
  "ABCDEFGHJKLMNPQRSTUVWXYZabcdefghjkmnpqrstuvwxyz23456789";
const TOKEN_LENGTH = 16;

export function generatePlaylistInviteToken(): string {
  const bytes = new Uint8Array(TOKEN_LENGTH);
  if (typeof crypto !== "undefined" && crypto.getRandomValues) {
    crypto.getRandomValues(bytes);
  } else {
    for (let i = 0; i < TOKEN_LENGTH; i += 1) {
      bytes[i] = Math.floor(Math.random() * 256);
    }
  }

  let token = "";
  for (let i = 0; i < TOKEN_LENGTH; i += 1) {
    token += TOKEN_ALPHABET[bytes[i]! % TOKEN_ALPHABET.length];
  }
  return token;
}

export function normalizeInviteToken(raw: string): string {
  return raw.trim();
}

File: mobile/lib/domain/playlist_invite_token.dart
import 'dart:math';

const _tokenAlphabet =
    'ABCDEFGHJKLMNPQRSTUVWXYZabcdefghjkmnpqrstuvwxyz23456789';
const int playlistInviteTokenLength = 16;
const int minInviteTokenLength = 12;

final _random = Random.secure();

String generatePlaylistInviteToken() {
  final buffer = StringBuffer();
  for (var i = 0; i < playlistInviteTokenLength; i++) {
    buffer.write(_tokenAlphabet[_random.nextInt(_tokenAlphabet.length)]);
  }
  return buffer.toString();
}

String normalizeInviteToken(String raw) => raw.trim();
